### PR TITLE
BSDK-362 Enable value transfer using executeBatch

### PR DESCRIPTION
**Pull request by AmanRaj1608**
- pass array of values in executeBatch function and remove hardcoded 0 in _call() function. use values[i] instead.

- rename execute method name which conflicts the internal execute method in Executor.sol

- while doing this, also remove onlyOwner modifier on execute() and executeBatch() functions(search 495 - execute() and executeBatch() functions in SmartAccount.sol from the EntryPoint will fail ) in mid risk page

- also combining this here: #313 making sure above functions and any other function that can have reentrancy uses non-reentrant modifier.

### CODE REVIEW

File: contracts/smart-contract-wallet/SmartAccount.sol [hunk 1]
```diff
@@ -463,16 +463,25 @@ contract SmartAccount is
...
+    function executeBatchCall(
+        address[] calldata dest,
+        uint256[] calldata value,
+        bytes[] calldata func
+    ) external nonReentrant {
         _requireFromEntryPointOrOwner();
-        require(dest.length == func.length, "wrong array lengths");
-        for (uint i = 0; i < dest.length;) {
-            _call(dest[i], 0, func[i]);
+        require(dest.length == value.length, "wrong array lengths");
```
Comment (filmakarov) [inline contracts/smart-contract-wallet/SmartAccount.sol:481]: should we explicitly revert when lengths are 0?

Comment (AmanRaj1608) [reply contracts/smart-contract-wallet/SmartAccount.sol:482]: resolved

Comment (AmanRaj1608) [reply contracts/smart-contract-wallet/SmartAccount.sol:482]: have removed and added a test

**Review by livingrockrises — APPROVED** [review]
LGTM

**Pull request merged**